src/pip/_internal/req/req_install.py

```diff
@@ -188,7 +189,7 @@ def __init__(
 
     def __str__(self) -> str:
         if self.req:
-            s = str(self.req)
+            s = redact_auth_from_requirement(self.req)
```

Comment:
Note to reviewers: this specific change is unrelated to the bug fix, but this should prevent similar issues popping up elsewhere.